### DISCUSSION TITLE
change the position of the paper elements dynamically

### DISCUSSION THREAD

**Discussion by slvamegestic** [clientIO/joint#2013] [Q&A] [answered]
Hi,

    I am using jointJs 3.6.5. 
 While the browser window is resized, I would like to change the position of the paper elements dynamically. Is there a way to accomplish this? 

I tried setting the element's position based on the window.innerWidth on the window `resize` event listener. However, I have failed.

In addition, I would like to achieve the width and height of the paper background image based on window.  Please let me know if you have any ideas for this one as well.

**Comment by kumilingus** [comment, marked as answer]
You can either scale down the paper/views (see [example](https://codepen.io/jointjs/pen/oNMeBGd)):
```js
window.on('resize', () => {
   paper.scaleContentToFit({ /* options */ });
});
```

Or reorganize the models in the graph:
```js
window.on('resize', () => {
  const area = paper.getArea(); // in graph coordinates
  el1.position(area.x, area.y); // top left corner
  el2.position(area.x + area.width - el2.size().width, 0); // top right corner
});
```

**Reply by slvamegestic** [reply]
thanks @kumilingus   I've tried this and it works. However, it is incompatible with the Background Image while I am resizing the window.
   The idea I tried to implement was...I want the elements to stay inside the background image no matter how small or large the window is.

This is how the background image has been appended.
` paper.drawBackground({
      image:'Image URL.svg',
      position: "top center",
       size: "contain",
     });`
Is there any property or idea that can be implemented to achieve this goal?

**Reply by kumilingus** [reply]
Please create a JSFiddle with what you just described.

**Reply by kumilingus** [reply]
Thank you for providing an example. It always helps.

Here's a working [JSFiddle](https://jsfiddle.net/kumilingus/psj47onL/).

A few notes:
- You need to define the background image with a **point** & **dimension** if you want the image to be transformed by the paper scale.
```js
paper.drawBackground({
    image:'https://www.linkpicture.com/q/Trolly_height-upd-Hand-MD_1.svg',
    position: { x: area.x, y: area.y },
    size: { width: area.width, height: area.height },
});
```
- You could have also add the image the same way you have added your rectangles (to simplify things, using the same approach)
```js
V('image', { `xlink:href`: 'image.svg', x, y, width, height }).addTo(paper.viewport);
```
- Draw the trolley and the racks first, then you scale down/up the paper so it covers entire screen.

**Reply by slvamegestic** [reply]
Thanks @kumilingus It's awesome. I'll take a look at them

**Reply by slvamegestic** [reply]
@kumilingus your suggestion is worked for me. Thanks a lot .

  i have set the background image as window width and height like below
 
```
const area = g.Rect(0, 0, window.innerWidth, window.innerHeight);
paper.drawBackground({
    image:'https://www.linkpicture.com/q/Trolly_height-upd-Hand-MD_1.svg',
    position: { x: area.x, y: area.y },
    size: { width: area.width, height: area.height },
});
```


Any suggestion to get the x , y position of Rects to set inside the image border?
See the updated fiddle [here](https://jsfiddle.net/selvakkumar/7pvo1qn6/31/)

**Reply by kumilingus** [reply]
It depends on the trolley image. If you use a different image, the position will be different.

**Reply by slvamegestic** [reply]
According to my understanding, we must manually set the x and y positions for each trolley image.?

**Reply by kumilingus** [reply]
For each trolley image, you must use a specific `x`, `y`, `width` and `height` on all rectangles.

**Reply by slvamegestic** [reply]
Ok Noted on that @kumilingus
